hyp3lib/get_orb.py

```diff
@@ -1,197 +1,140 @@
-  orbitFile,url = getPageContentsESA(q, False)
-  if url:
-    print("Using url {}".format(url))
-  else:
-    print("Unable to find POEORB - Looking for RESORB")
-    q=restituted+'validity_stop__gt='+start_time.strftime('%Y-%m-%dT%H:%M:%S')+'&validity_start__lt='+end_time.strftime('%Y-%m-%dT%H:%M:%S')
-    orbitFile,url = getPageContentsESA(q, False)
-    if url:
-      print("Using url {}".format(url))
-    else:
-      error = 'Could not find orbit file on ESA website'
-      raise FileException(error)
-
-  return url, orbitFile
-
-
-def fetchOrbitFile(urlOrb,stateVecFile,verify):
-  hostname = urlparse(urlOrb).hostname
-  session = requests.Session()
-  session.mount(hostname, HTTPAdapter(max_retries=10))
-  request = session.get(urlOrb, timeout=60, verify=verify)
-  f = open(stateVecFile, 'w')
-  f.write(request.text)
-  f.close()
-
-
-def downloadSentinelOrbitFileProvider(granule, provider, directory):
-  if provider.upper() == 'ASF':
-    urlOrb, fileNameOrb = getOrbFile(granule)
-    verify = True
-  elif provider.upper() == 'ESA':
-    urlOrb, fileNameOrb = getOrbitFileESA(granule)
-    verify = False
-
-  if directory:
-    stateVecFile = os.path.join(directory, fileNameOrb)
-  else:
-    stateVecFile = fileNameOrb
-
-  if not os.path.isfile(stateVecFile):
-    if len(stateVecFile) > 0:
-      fetchOrbitFile(urlOrb,stateVecFile,verify)
-      return stateVecFile
+    return search_url + best
+
+
+def get_orbit_url(granule: str, orbit_type: str = 'AUX_POEORB', provider: str = 'ESA'):
+    """Get the URL of an orbit file from a provider
+
+    Args:
+        granule: Granule name to find an orbit file for
+        orbit_type: Orbit type to download
+        provider: Povider name to download the orbit file from
+
+    Returns:
+        orbit_url: The url to the matched orbit file
+    """
+    platform = granule[0:3]
+    time_stamps = re.split('_+', granule)[4:6]
+
+    if provider == 'ESA':
+        delta = timedelta(seconds=60)
+        start_time = datetime.strptime(time_stamps[0], '%Y%m%dT%H%M%S') - delta
+        end_time = datetime.strptime(time_stamps[1], '%Y%m%dT%H%M%S') + delta
+
+        params = {
+            "product_type": orbit_type.upper(),
+            "product_name__startswith": platform,
+            "validity_start__lt": start_time.strftime('%Y-%m-%dT%H:%M:%S'),
+            "validity_stop__gt": end_time.strftime('%Y-%m-%dT%H:%M:%S'),
+            "ordering": "-creation_date",
+            "page_size": "1",
+        }
+
+        response = requests.get(url='https://qc.sentinel1.eo.esa.int/api/v1/', params=params)
+        response.raise_for_status()
+        qc_data = response.json()
+
+        orbit_url = None
+        if qc_data["results"]:
+            orbit_url = qc_data["results"][0]["remote_url"]
+        return orbit_url
+
+    elif provider.upper() == 'ASF':
+        url = f'https://s1qc.asf.alaska.edu/{orbit_type.lower()}/'
```

Comment:
move to `_get_asf_orbit_url`, I should have to tell something called `get_asf_orbit` where asf lives